client/src/components/CheckoutOptions.jsx

```diff
@@ -1,20 +1,69 @@
+/* eslint-disable arrow-body-style */
+/* eslint-disable no-useless-constructor */
+/* eslint-disable class-methods-use-this */
 import React from 'react';
 
 
 class CheckoutOptions extends React.Component {
   constructor(props) {
     super(props);
+
+    this.state = {
+      colorModifier: 1,
+      quantity: 1,
+    };
+
+    this.handleColorChoice = this.handleColorChoice.bind(this);
+    this.handleQuantity = this.handleQuantity.bind(this);
   }
 
+  handleColorChoice(e) {
+    this.setState({
+      colorModifier: Number(e.target.value),
+    }, () => {
+      this.props.handleColorChoice(this.state.colorModifier);
+    });
```

Comment:
i wonder what this would be like:
```
const colorModifier = Number(e.target.value);
this.setState({ colorModifier });
this.props.handleColorChoice(colorModifier);
```